apps/prairielearn/src/lib/socket-server.js

```diff
@@ -1,67 +1,58 @@
 // @ts-check
 const { Server } = require('socket.io');
-const redis = require('redis');
+const { Redis } = require('ioredis');
 const { createAdapter } = require('@prairielearn/socket.io-redis-adapter');
 const { logger } = require('@prairielearn/logger');
 const path = require('path');
 const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
 
 const { config } = require('./config');
 
+/**
+ * @param {Redis} client
+ * @param {string} type
+ */
+function attachEventListeners(client, type) {
+  client.on('error', (err) => {
+    logger.error(`redis ${type} client: error`, err);
```

Comment:
Can we make this a bit easier to grep for `redis client`:
```suggestion
    logger.error(`redis client ${type}: error`, err);
```